br/pkg/restore/split_test.go

```diff
@@ -560,3 +565,123 @@ func TestRegionConsistency(t *testing.T) {
 		require.Regexp(t, ca.err, err.Error())
 	}
 }
+
+type fakeRestorer struct {
+	errorInSplit  bool
+	splitRanges   []rtree.Range
+	restoredFiles []*backuppb.File
+}
+
+func (f *fakeRestorer) SplitRanges(ctx context.Context, ranges []rtree.Range, rewriteRules *restore.RewriteRules, updateCh glue.Progress, isRawKv bool) error {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+	f.splitRanges = append(f.splitRanges, ranges...)
+	if f.errorInSplit {
+		err := errors.Annotatef(berrors.ErrRestoreSplitFailed,
+			"the key space takes many efforts and finally get together, how dare you split them again... :<")
+		log.Error("error happens :3", logutil.ShortError(err))
+		return err
+	}
+	return nil
+}
+
+func (f *fakeRestorer) RestoreFiles(ctx context.Context, files []*backuppb.File, rewriteRules *restore.RewriteRules, updateCh glue.Progress) error {
+	if ctx.Err() != nil {
```

Comment:
🤔 do we have similar code in real RestoreFiles?

Reply:
Probably, but hidden in some internal call of the clients.

Reply:
For example, here:
<img width="798" alt="image" src="https://user-images.githubusercontent.com/36239017/158781437-1ca16d59-80e7-422c-9dfa-89b3e5cf1eea.png">